modules/gdscript/gdscript_parser.cpp

```diff
@@ -642,6 +642,52 @@ void GDScriptParser::parse_program() {
 	clear_unused_annotations();
 }
 
+GDScriptParser::ClassNode *GDScriptParser::find_class(const String &p_qualified_name) const {
+	String first = p_qualified_name.get_slice("::", 0);
+
+	Vector<String> class_names;
+	GDScriptParser::ClassNode *result = nullptr;
+	// Empty initial name means start at the head.
+	if (first.is_empty() || (head->identifier && first == head->identifier->name)) {
+		class_names = p_qualified_name.split("::");
+		result = head;
+	} else if (p_qualified_name.begins_with(script_path)) {
+		// Script path could have a class path separator("::") in it.
+		class_names = p_qualified_name.trim_prefix(script_path).split("::");
+		result = head;
+	} else if (head->has_member(first)) {
+		class_names = p_qualified_name.split("::");
+		GDScriptParser::ClassNode::Member member = head->get_member(first);
+		if (member.type == GDScriptParser::ClassNode::Member::CLASS) {
+			result = member.m_class;
+		}
+	}
+
+	for (int i = 1; result != nullptr && i < class_names.size(); i++) {
```

Comment:
`// Starts at 1 because the 0 is `first` which was handled above.`

It really throws me when `for` loops don't start at 0 :) in this case, it also works even if `first == nullptr` because in that case, class_names has a single element, and therefore `1 < 1` fails and the loop doesn't run.

It all checks out! <3

Reply:
ya, it does look a bit confusing, ill add a comment. thanks!